blivet/static_data/nvdimm.py

```diff
@@ -0,0 +1,129 @@
+#
+# nvdimm.py - nvdimm class
+#
+# Copyright (C) 2018  Red Hat, Inc.  All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import gi
+gi.require_version("BlockDev", "2.0")
+from gi.repository import BlockDev
+
+import logging
+log = logging.getLogger("blivet")
+
+
+class NVDIMM(object):
+    """ NVDIMM utility class.
+
+        .. warning::
+            Since this is a singleton class, calling deepcopy() on the instance
+            just returns ``self`` with no copy being created.
+    """
+
+    def __init__(self):
+        self._namespaces = None
+
+    # So that users can write nvdimm() to get the singleton instance
+    def __call__(self):
+        return self
+
+    def __deepcopy__(self, memo_dict):
+        # pylint: disable=unused-argument
+        return self
+
+    @property
+    def namespaces(self):
+        """ Dict of all NVDIMM namespaces, including dax and disabled namespaces
+        """
+        if not self._namespaces:
+            self.update_namespaces_info()
+
+        return self._namespaces
+
+    def update_namespaces_info(self):
```

Comment:
Shouldn't this use the [``DependencyGuard``](https://github.com/storaged-project/blivet/blob/2.1-devel/blivet/util.py#L1073) or some other on-demand availability checking mechanism?

Reply:
Good idea, fixed.